site/server/test/PersistenceModuleSpec.scala

```diff
@@ -0,0 +1,454 @@
+  val insertSql = "INSERT INTO persistence (name,active) VALUES (?,?)"
+  val updateAllSql = "UPDATE persistence SET active=false"
+  val updateAllActiveSql = "UPDATE persistence SET active=false WHERE active=true"
+  val updateByIdSql = "UPDATE persistence SET name=?,active=? WHERE id=?"
+  val updateByStatusSql = "UPDATE persistence SET active=? WHERE active=?"
+
+  def createTable: ConnectionIO[Int] =
+    sql"""
+          CREATE TABLE persistence (
+          id   BIGINT AUTO_INCREMENT,
+          name VARCHAR NOT NULL,
+          active BOOL NOT NULL)""".update.run
+
+  def dropTable: ConnectionIO[Int] = sql"""DROP TABLE IF EXISTS persistence""".update.run
+
+  def insertItem(
+    name:   String,
+    active: Boolean
+  ): ConnectionIO[Long] =
+    sql"INSERT INTO persistence (name, active) VALUES ($name,$active)".update.withUniqueGeneratedKeys[Long]("id")
+
+  def insertItems(
+    values: List[(String, Boolean)]
+  ): ConnectionIO[Int] =
+    Update[(String, Boolean)]("INSERT INTO persistence (name, active) VALUES (?,?)").updateMany(values)
+
+  def fetchAll: ConnectionIO[List[(String, Boolean, Long)]] =
+    sql"SELECT name,active,id FROM persistence".query[(String, Boolean, Long)].list
+
+  def fetchItemById(
+    id: Long
+  ): ConnectionIO[PersistenceItem] =
+    sql"SELECT id,name,active FROM persistence WHERE id=$id".query[PersistenceItem].unique
+
+  def fetchItemByStatus(
+    active: Boolean
+  ): ConnectionIO[List[PersistenceItem]] =
+    sql"SELECT id,name,active FROM persistence WHERE active=$active".query[PersistenceItem].list
+
+  def fetchItemByStatuses(
+    active: Boolean
+  ): ConnectionIO[List[PersistenceItem]] = {
+    val inactive = !active
+    sql"SELECT id,name,active FROM persistence WHERE active=$active OR active=$inactive".query[PersistenceItem].list
+  }
+
+  val persistenceModule = new PersistenceModule
+
+  def genBoundedList[T](minSize: Int = 1, maxSize: Int = 100, gen: Gen[T]): Gen[List[T]] =
+    Gen.choose(minSize, maxSize) flatMap { size ⇒ Gen.listOfN(size, gen) }
+
+  implicit val dataWithId: Arbitrary[List[(Long, String, Boolean)]] =
+    Arbitrary(
+      genBoundedList(
+        minSize = 2,
+        maxSize = 10,
+        gen = Gen.resultOf((l: Long, s: String, b: Boolean) ⇒ (l, s, b))
+      )
+    )
+
+  implicit val dataStringBoolean: Arbitrary[List[(String, Boolean)]] =
+    Arbitrary(genBoundedList(
+      minSize = 2,
+      maxSize = 10,
+      gen = Gen.resultOf((s: String, b: Boolean) ⇒ (s, b))
+    ))
+
+  implicit val dataStringBooleanLong: Arbitrary[List[(String, Boolean, Long)]] =
+    Arbitrary(genBoundedList(
+      minSize = 2,
+      maxSize = 10,
+      gen = Gen.resultOf((s: String, b: Boolean, l: Long) ⇒ (s, b, l))
+    ))
+
+  implicit val stringList: Arbitrary[List[String]] =
+    Arbitrary(genBoundedList(minSize = 2, gen = Gen.resultOf((s: String) ⇒ s)))
+
+  implicitly[Arbitrary[PersistenceItem]]
+
+  val persistenceItemGen = for {
```

Comment:
Should this just define the arbitrary instance instead of the generator to be consistent with the rest?